soapui/src/main/java/com/eviware/soapui/support/swing/FunctionalMenu.java

```diff
@@ -0,0 +1,139 @@
+
+import com.eviware.soapui.SoapUI;
+import com.eviware.soapui.impl.wsdl.WsdlProject;
+import com.eviware.soapui.impl.wsdl.WsdlTestSuite;
+import com.eviware.soapui.impl.wsdl.actions.support.ShowDesktopPanelAction;
+import com.eviware.soapui.impl.wsdl.actions.teststep.ToggleDisableTestStepAction;
+import com.eviware.soapui.impl.wsdl.testcase.WsdlTestCase;
+import com.eviware.soapui.impl.wsdl.teststeps.WsdlTestStep;
+import com.eviware.soapui.model.ModelItem;
+import com.eviware.soapui.model.support.ModelSupport;
+import com.eviware.soapui.model.tree.SoapUITreeNode;
+import com.eviware.soapui.model.workspace.Workspace;
+import com.eviware.soapui.support.action.SoapUIActionMapping;
+import com.eviware.soapui.support.action.support.DefaultActionMapping;
+import com.eviware.soapui.support.action.swing.ActionList;
+import com.eviware.soapui.support.action.swing.ActionListBuilder;
+import com.eviware.soapui.support.action.swing.ActionSupport;
+import com.eviware.soapui.support.action.swing.SwingActionDelegate;
+import javax.swing.*;
+import java.awt.*;
+
+
+public class FunctionalMenu {
+    public static JMenu getMenu(String name){
+        for (int i = 0; i<SoapUI.getMenuBar().getMenuCount(); i++){
+            if (SoapUI.getMenuBar().getMenu(i).getText().equals(name)){
+                return SoapUI.getMenuBar().getMenu(i);
+            }
+        }
+        return null;
+    }
+
+    public static JMenu buildMenuForWorkspace(JMenu menu, String actionGroup) {
+        ActionList actions = buildActionsForActionGroup(actionGroup);
+        if(menu.getText().equals(SoapUI.STEP)){
+            SoapUIActionMapping<WsdlTestStep> toggleDisabledActionMapping = null;
+            DefaultActionMapping<WsdlTestStep> actionMapping = new DefaultActionMapping<WsdlTestStep>(
+                        ShowDesktopPanelAction.SOAPUI_ACTION_ID, "ENTER", null, true, null);
+            actionMapping.setName("Open Editor");
+            actionMapping.setDescription("Opens the editor for this TestStep");
+            SwingActionDelegate actionDelegate = new SwingActionDelegate(actionMapping, null);
+            menu.add(actionDelegate);
+
+            toggleDisabledActionMapping = new DefaultActionMapping<WsdlTestStep>(
+                    ToggleDisableTestStepAction.SOAPUI_ACTION_ID, null, null, false, null);
+
+            SwingActionDelegate actionDelegateToggle = new SwingActionDelegate(toggleDisabledActionMapping, null);
+            menu.add(actionDelegateToggle);
+        }
+        ActionSupport.addActions(actions, menu);
+        for (Component component : menu.getMenuComponents()) {
+            disableEnableMenuSubItems(component, false);
+            component.setEnabled(false);
+        }
+        return menu;
+    }
+
+    protected static ActionList buildActionsForActionGroup(String actionGroup) {
+        return ActionListBuilder.buildActions(actionGroup, null);
+    }
+
+    private static void disableMenuItems(String groupId) {
```

Comment:
ooooops, disableMenuItems and enableMenuItems should be replaced with one function activateMenuItems (String groupId, boolean activate). See DRY principle https://ru.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself